### PR TITLE
add DUNE scripts

### PR DESCRIPTION
**Pull request by Raye174**

### CODE REVIEW

File: strings.conf [hunk 1]
```diff
@@ -5,6 +5,7 @@
 #counters
 !counter 0x1162 Bunny Ears Counter
 !counter 0x163 Counter(Gunkan Suship Chef's Choice)
+!counter 0x65 Signal Counter
```
Comment (mercury233) [inline strings.conf:8]: Signal Counter is `0x104d` and is used by _Signal Warrior_ and _On Your Mark, Get Set, DUEL!_ already

Comment (Raye174) [reply strings.conf:8]: haii

File: scripts/DUNE-JP/c101201204.lua [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+		Duel.SendtoHand(g,nil,REASON_EFFECT)
+		Duel.ConfirmCards(1-tp,g)
+	end
+end
+function s.cfilter(c,e,tp)
+	return c:IsFaceup() and c:IsLevelAbove(7) and c:IsType(TYPE_SYNCHRO) and not c:IsCode(id)
+		and Duel.IsExistingMatchingCard(s.spfilter,tp,LOCATION_EXTRA,0,1,nil,e,tp,c:GetLevel())
+end
+function s.spfilter(c,e,tp,lv)
+	return c:IsLevelAbove(lv) and c:IsType(TYPE_SYNCHRO)
```
Comment (Lyris12) [inline scripts/DUNE-JP/c101201204.lua:54]: > (Quick Effect): You can target 1 Level 7 or higher Synchro Monster on the field, except “The Crimson Dragon”; return this card to the Extra Deck, and if you do, Special Summon 1 Dragon Synchro Monster with the same Level from your Extra Deck. (This is treated as a Synchro Summon.)
```suggestion
	return c:IsLevel(lv) and c:IsType(TYPE_SYNCHRO)
```

**Review by Lyris12 — CHANGES_REQUESTED** [review]

File: scripts/DUNE-JP/c101201204.lua (hunk 2 above)
Comment (Raye174) [reply scripts/DUNE-JP/c101201204.lua:54]: i didn't double check it properly thank you for notice this

File: scripts/DUNE-JP/c101201204.lua [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+	end
+end
+function s.cfilter(c,e,tp)
+	return c:IsFaceup() and c:IsLevelAbove(7) and c:IsType(TYPE_SYNCHRO) and not c:IsCode(id)
+		and Duel.IsExistingMatchingCard(s.spfilter,tp,LOCATION_EXTRA,0,1,nil,e,tp,c:GetLevel())
+end
+function s.spfilter(c,e,tp,lv)
+	return c:IsLevel(lv) and c:IsType(TYPE_SYNCHRO)
+		and Duel.GetLocationCountFromEx(tp,tp,nil,c)>0
+		and c:IsCanBeSpecialSummoned(e,SUMMON_TYPE_SYNCHRO,tp,false,false)
```
Comment (Lyris12) [inline scripts/DUNE-JP/c101201204.lua:56]: > (Quick Effect): You can target 1 Level 7 or higher Synchro Monster on the field, except “The Crimson Dragon”; return this card to the Extra Deck, and if you do, Special Summon 1 **Dragon** Synchro Monster with the same Level from your Extra Deck. (This is treated as a Synchro Summon.)

```suggestion
	return c:IsLevel(lv) and c:IsType(TYPE_SYNCHRO) and c:IsRace(RACE_DRAGON)
		and Duel.GetLocationCountFromEx(tp,tp,nil,c)>0
		and c:IsCanBeSpecialSummoned(e,SUMMON_TYPE_SYNCHRO,tp,false,false)
```

**Review by Lyris12 — CHANGES_REQUESTED** [review]

File: scripts/DUNE-JP/c101201205.lua [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+end
+function s.cfilter(c,e,tp)
+	return s.spfilter(c,e,tp) and c:IsCanBeEffectTarget(e)
+end
+function s.target(e,tp,eg,ep,ev,re,r,rp,chk,chkc)
+	if chkc then return chkc:IsLocation(LOCATION_GRAVE) and chkc:IsControler(tp) and s.spfilter(chkc,e,tp) end
+	local ft=Duel.GetLocationCount(tp,LOCATION_MZONE)
+	local cg=Duel.GetMatchingGroupCount(s.cfilter,tp,LOCATION_GRAVE,0,nil,e,tp)
+	if chk==0 then return cg>=ft and not Duel.IsPlayerAffectedByEffect(tp,59822133) end
+	local ct=math.min(cg,ft)
```
Comment (Lyris12) [inline scripts/DUNE-JP/c101201205.lua:43]:
```suggestion
	if chk==0 then return ft>0 end
	local cg=Duel.GetMatchingGroupCount(s.cfilter,tp,LOCATION_GRAVE,0,nil,e,tp)
	local ct=math.min(cg,ft)
	if Duel.IsPlayerAffectedByEffect(tp,59822133) then ct=1 end
```

**Review by Lyris12 — CHANGES_REQUESTED** [review]
Bug: Cannot activate to Summon 2 Level 8 Dragon Synchro monsters in GY into 4 Main Monster Zones

**Pull request merged**